iosApp/iosApp/ContentViewController.swift

```diff
@@ -322,6 +325,15 @@ extension ContentViewController : UITextFieldDelegate {
             })
         case ("clearConversation", _):
             client.invalidateConversationCache()
+        case ("addAttribute", let msg?):
+            let segments = splitUserInput(msg)
+            if((segments.0?.isEmpty) == nil) {
```

Comment:
if `segments.0` is an empty string here this will still fall through to the `else` clause. I'd recommend restructuring like the following:

```
case ("addAttribute", let msg?):
    let segments = splitUserInput(msg)
    if let key = segments.0, !key.isEmpty {
        customAttributes[key] = segments.1 // is it ok for segments.1 (the value) to be nil or should it be checks in the if clause too?
        self.info.text = "Custom attribute added: \(segments)"
    }  else {
        self.info.text = "Custom attribute key cannot be null or empty!"
    }
```

Reply:
@crumpf Thanks for feedback. 
I have updated the code. And no `segment. 1` should not be nil. So I added a check to fallback to empty string if it is.